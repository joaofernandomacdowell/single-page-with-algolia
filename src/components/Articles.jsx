import React, { useState } from 'react';
import algoliasearch from 'algoliasearch/lite';
import {
  InstantSearch,
  SearchBox,
  Pagination,
  connectHits,
} from 'react-instantsearch-dom';

import ArticleItem from './ArticleItem';
import ArticleFilter from './ArticleFilter';
import Header from './Header';


const APP_ID = process.env.REACT_APP_ID;
const API_KEY = process.env.REACT_APP_API_KEY;

const searchClient = algoliasearch(APP_ID, API_KEY);

const renderHits = (hits) =>
  hits ? hits.map(hit => <ArticleItem hit={hit} key={hit.id} />) : null;

const ArticleItems = ({ hits }) => (
  <div className="zoom-articles-container">
    {renderHits(hits)}
  </div>
);

const CustomHits = connectHits(ArticleItems);

const Articles = () => {
  const [isClosed, setClose] = useState(true);

  return (
    <InstantSearch
      searchClient={searchClient}
      indexName="challenge_article_v2_items">
    <Header>
      <SearchBox translations={{ placeholder: 'Digite sua busca...'}} />
      <ArticleFilter isClosed={isClosed} setClose={setClose} />
    </Header>
    <CustomHits />
    <Pagination
      showFirst={false}
    />
    </InstantSearch>
  );
};

export default Articles;