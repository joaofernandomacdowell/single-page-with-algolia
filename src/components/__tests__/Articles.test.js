import React from 'react'
import { shallow } from 'enzyme';

import mockApiResponse from '../../helper';
import Articles from '../Articles';


describe('Articles component', () => {
  const props = {
    title: mockApiResponse.hits[0].title
  };
  const wrapper = shallow(<Articles {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});