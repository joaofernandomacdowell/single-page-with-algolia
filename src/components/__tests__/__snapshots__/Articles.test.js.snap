// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component match the snapshot 1`] = `ShallowWrapper {}`;
